app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -43,35 +43,25 @@ class SchoolDisciplineDashboard extends React.Component {
     });
   }
 
-  studentDisciplineIncidentCounts(incidentCategory) {
+  studentDisciplineIncidentCounts(incidents) {
     let studentDisciplineIncidentCounts = {};
-    const selectedChartData = this.getChartData(this.state.selectedChart);
-    const incidents = incidentCategory ?
-                      selectedChartData.disciplineIncidents[incidentCategory] :
-                      this.filterIncidentDates(this.props.schoolDisciplineEvents);
```

Comment:
all filtering now happens once in the render method and passed both here and the chart